src/hdx_cli_toolkit/utilities.py

```diff
@@ -335,3 +335,31 @@ def print_dictionary_comparison(
                 flush=True,
             )
     print("-" * total_width, flush=True)
+
+
+def query_dict(
+    key_: str,
+    output: list[dict[str, Any]],
+    dataset_dict: dict[str, Any],
+    output_row: dict[str, Any],
+) -> list[dict[str, Any]]:
+    if "." not in key_:
+        output_row[key_] = dataset_dict.get(key_, "Key absent")
+        output.append(output_row)
+    else:
+        key1, key2 = key_.split(".")
+        intermediate_value = dataset_dict.get(key1, "Key1 absent")
+        if isinstance(intermediate_value, dict):
+            output_row[key_] = intermediate_value.get(key2, "Key2 absent")

```

Comment:
What about recursion to handle more depth?

Reply:
Automating the with_extras thing means I don't need to consider this. I think on the whole I prefer to have a consistent name rather than a magic conversion to a legal Python name

Reply:
Recursion scares me! I might give this a go, limiting to 2 keys deep at least means we don't have to handle nasty things like list.list.key

Reply:
I implemented this to satisfy my curiosity, I don't think I'll deploy although it sort of handles lists it does not handle nested lists and it isn't clear how you would provide the the "prefix" information from non-list columns. See the `traverse` functions and its tests.